### PR TITLE
vtgate buffer: Add test when failover is done externally and TabletExternallyReparented is called.

### PR DESCRIPTION
**Pull request by michael-berlin**

### CODE REVIEW

File: test/tablet.py [hunk 1]
```diff
@@ -476,6 +479,8 @@ def start_vttablet(
     args.extend(['-watch_replication_stream'])
     if enable_semi_sync:
       args.append('-enable_semi_sync')
+      # Remember the setting in case a test wants to know it.
+      self.enable_semi_sync = enable_semi_sync
```
Comment (alainjobart) [inline test/tablet.py]: SHouldn't this be set in any case, not in the if? i.e. indent 2 spaces left?

File: test/mysql_flavor.py [hunk 2]
```diff
@@ -139,25 +147,28 @@ def change_master_commands(self, host, port, pos):
...
   """Set the object that will be returned by mysql_flavor().
 
   If flavor is not specified, set it based on MYSQL_FLAVOR environment variable.
+
+  Args:
+    flavor: String of the MySQL flavor e.g. "MariaDB" or "MySQL56".
   """
 
-  global __mysql_flavor
+  global MYSQL_FLAVOR
```
Comment (alainjobart) [inline test/mysql_flavor.py]: is that the standard for python, global variables in capital?

Comment (michael-berlin) [inline test/mysql_flavor.py]: The linter thinks it's a constant and therefore it wants it all caps. (Constants must be written all caps: https://www.python.org/dev/peps/pep-0008/#constants)

File: test/tablet.py (hunk 1 above)
Comment (michael-berlin) [inline test/tablet.py]: Yes, good point.

Done.

**Pull request merged**